src/main/java/org/kuali/kfs/coa/document/validation/impl/AccountGlobalRule.java

```diff
@@ -863,5 +962,1065 @@ private boolean isIncomeStreamAccountRequired(String fundGroupCode, String subFu
         return required;
     }
 
+    protected boolean checkSubFundProgram(MaintenanceDocument document) {
+        boolean success = true;
+
+        String subFundGroupCode = newAccountGlobal.getSubFundGroupCode();
+        String subFundProg = newAccountGlobal.getProgramCode();
+        BusinessObjectService bos = SpringContext.getBean(BusinessObjectService.class);
+
+        if (!StringUtils.isBlank(subFundProg)) {
+            Map fieldValues = new HashMap();
+            fieldValues.put(KFSPropertyConstants.SUB_FUND_GROUP_CODE, subFundGroupCode);
+            fieldValues.put(CUKFSPropertyConstants.PROGRAM_CODE, subFundProg);
+            
+            Collection<SubFundProgram> retVals = bos.findMatching(SubFundProgram.class, fieldValues);
+            
+            if (retVals.isEmpty()) {
+                success = false;
+                putFieldError(CUKFSPropertyConstants.PROGRAM_CODE, CUKFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_PROGRAM_CODE_NOT_GROUP_CODE, new String[] {subFundProg, subFundGroupCode});
+            } else {
+            	for (SubFundProgram sfp : retVals) {
+            		if (!sfp.isActive()) {
+                        putFieldError(CUKFSPropertyConstants.PROGRAM_CODE, KFSKeyConstants.ERROR_INACTIVE, getFieldLabel(Account.class, CUKFSPropertyConstants.PROGRAM_CODE));
+                        success = false;
```

Comment:
Given that the error message is not dependent on the SubFundProgram field values, it might be good to short-circuit this loop after finding an inactive one so as to avoid potentially adding duplicate messages for multiple inactive ones.